### ISSUE TITLE
Pre/Post steps broke windows support

### ISSUE THREAD

**Issue by bouskaJ** [citrusframework/yaks#112]
We agreed in issues #55 to have the following format for `yaks-config` files:

```
config:
  some-general-config: tbd
pre:
- script: ../prepare.sh
- run: |
    kamel version

    kamel install
    kamel run xxx.groovy
post:
- run: echo "Hello!"
```
https://github.com/citrusframework/yaks/blob/master/pkg/cmd/test.go#L580 wraps multiline commands to the `yaks-script-*.sh` file.

This approach breaks windows support because windows is not able to execute `.sh` file with `#!/bin/bash` (https://github.com/citrusframework/yaks/blob/master/pkg/cmd/test.go#L586) directive.

Maybe we should reconsider our decision with bash scripts and go on the Tekton pipeline way as @christophd suggested.

**Comment by bouskaJ** [comment]
@nicolaferraro WDYT? do you see any other option?

**Comment by nicolaferraro** [comment]
> @nicolaferraro WDYT? do you see any other option?

I see. From one side, the multiline scripts are handy and are executed as expected in linux/mac, including all kubernetes based pipeline implementations that run on linux.

Also the `script` instruction is OS sensitive. I fear that also e.g. the `kamel` command is subject to issues on windows (it's called `kamel.exe` there).

Wondering if we can tell users to use a bash-compatible shell on windows...

**Comment by VratislavHais** [comment]
Hello @nicolaferraro,

Do you think it is a good idea to basically force customers to use bash-compatible shell on windows? I mean it would certainly make our lives easier but I don't think it's a good approach. We should be aiming for multiplatform support rather than "Make your windows more similar to Linux".

**Comment by nicolaferraro** [comment]
I see your point @VratislavHais. 

The purpose of those scripts is to allow to prepare the environment under test via external tools and they should work preferably without changes on any OS. But writing a layer on top of bash is not our goal... so we may try change the semantics of the "run" instruction from "write any bash" to "write a sequence of commands".

I'm not using Windows since ages, can a command like `kubectl get pod` be used on windows or it should be `kubect.exe`?

**Comment by VratislavHais** [comment]
@nicolaferraro I think it is possible in powershell. I'll check it first just to be sure.

**Comment by bouskaJ** [comment]
@nicolaferraro My point was that windows users will not be able to run for example this test https://github.com/openshift-integration/camel-k-example-vdb/blob/master/test/yaks-config.yaml because of https://github.com/openshift-integration/camel-k-example-vdb/blob/master/test/scripts/installDV.sh 

I can try to create multiplatform script (seqence of commands) but it is very tricky..

**Comment by nicolaferraro** [comment]
@bouskaJ I've seen that, do you strictly need the `waitFor` ? Maybe that can be replaced by some other commands..

I've also tried to reach the oc client folks and operator-sdk folks to propose a CLI to install operators more easily, but no luck so far..

**Comment by bouskaJ** [comment]
@nicolaferraro I think that the bigger pain is the `sed` command (not supported on windows) in `sed "s/YAKS_NAMESPACE/${YAKS_NAMESPACE}/" "$TEST_HOME"/resources/operatorGroup.yaml `

**Comment by VratislavHais** [comment]
@nicolaferraro so I have tried to execute the command as you've mentioned but only with oc. I have a file oc.exe and path have an entry pointing to the folder which contains oc.exe. I can execute `oc version` without specifying ".exe" suffix

**Comment by nicolaferraro** [comment]
> @nicolaferraro I think that the bigger pain is the `sed` command (not supported on windows) in `sed "s/YAKS_NAMESPACE/${YAKS_NAMESPACE}/" "$TEST_HOME"/resources/operatorGroup.yaml `

I think we can should avoid bash at all and rely on tools like https://github.com/kubernetes-sigs/kustomize for those cases.

I'm thinking to change the semantics of the run option to be a sequence of commands. I'm not sure on how to pass contextual env vars in windows...

**Comment by VratislavHais** [comment]
Hello @nicolaferraro , we have been discussing this issue with @bouskaJ today and we have a possible solution:
1) detect bash compatible shell installation (Cygwin) and run the script
2) detect Powershell installation
3) run the script in cmd

I'm gonna investigate it tomorrow so I'll have more information about our possibilities

**Comment by VratislavHais** [comment]
Hello @nicolaferraro . Forgive me I did not have time yesterday. I am checking it right now and there might not be any changes needed. I've found this: 
https://www.howtogeek.com/249966/how-to-install-and-use-the-linux-bash-shell-on-windows-10/

this features can be enabled really easily and it seems like it does contain even sed command. Since we newly have windows server 2019 (and for windows server it is available since 2019) I am gonna generate a new images with this feature and try whether everything works.
